### PR TITLE
feat: Add comment box and mandatory functionality NUM-18

### PR DESCRIPTION
**Pull request by tigs1995**

### CODE REVIEW

File: src/components/CommentComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  errorNoInput: boolean;
+};
+
+export default function CommentComponent({
+  onCommentChange,
+  errorNoInput
+}: CommentComponentProps) {
+
+  return (
+    <Box as="label" p="8">
```
[the remark this answers is not in the document]
Comment (djaustin) [reply src/components/CommentComponent.tsx:16]: I don't think this needs to be its own component. Feels like a premature abstraction that doesn't provide us any benefit.

File: src/components/CommentComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <Text>Add a comment</Text>
+      {errorNoInput ? 
+        <Input onChange={onCommentChange} isInvalid/>
```
Comment (djaustin) [inline src/components/CommentComponent.tsx:19]: Register this input with react-hook-form let it manage this stuff for you

File: src/components/FeedbackButton.tsx [hunk 3]
```diff
@@ -40,7 +43,7 @@ export default function FeedbackButton({
 
   return (
     <Box as="label">
-      <input {...getInputProps()} />
+      <input {...getInputProps()} onChange={onFeedbackChange}/>
```
Comment (djaustin) [inline src/components/FeedbackButton.tsx:46]: react-hook-form should handle all form state so we shouldn't have to add custom bits here to store the state of the score

File: src/pages/index.tsx [hunk 4]
```diff
@@ -6,24 +6,46 @@ import {
...
+  };
+
   const sendFeedback: SubmitHandler<FormInputs> = (data) => {
-    const parsedData = { ...data, score: parseInt(data.score) };
-    fetch("/api/feedback", {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify(parsedData),
-    });
+    if(feedbackValue == 1 && !comment){
```
Comment (djaustin) [inline src/pages/index.tsx:37]: the `handleSubmit` function on line 55 should be handling all our validation so we don't need to do it ourselves

**Review by djaustin — CHANGES_REQUESTED** [review]
This needs changes to the schema to make sure you can actually save this comment to the database

**Comment by djaustin** [comment]
Fix build issues form Vercel and code quality issues from codeclimate abov

File: src/components/CommentComponent.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tigs1995) [reply src/components/CommentComponent.tsx]: I removed the component in the end

**Review by djaustin — APPROVED** [review]

**Review by annabelkimber — APPROVED** [review]

**Pull request merged**